docs/10-certificate-key/03-installation-guide/02-database-setup.md

```diff
@@ -1,34 +1,61 @@
-:::
-
-## Setup database manually
-
-The database must be setup before the platform can be used. Depending on your setup, you should initialize the database using the following steps:
+There are several microservices that are using the database. Although each microservice can have its own database, it is recommended to use a single database for all microservices running in the same namespace. The platform will take care of creating all necessary schemas and tables itself in that case.
 
-- Create the database user
+Typicaly you need to create the database user for the platform:
 ```sql
 CREATE USER czertainlyuser WITH PASSWORD 'your-strong-password';
 ```
 
-- Create the database and grant the user access to it
+Create the database itself and grant the user full control over it:
 ```sql
 CREATE DATABASE czertainlydb ENCODING 'UTF8' LC_COLLATE='en_US.UTF-8' LC_CTYPE='en_US.UTF-8' TEMPLATE=template0;
 GRANT ALL PRIVILEGES ON DATABASE czertainlydb to czertainlyuser;
 ```
+The platform will take care of creating all necesary schemas and tables itself. It will also handle any necessary upgrades.
+
+:::warning
+Please note that downgrade of the platform is not supported. Once the database is upgraded, it cannot be downgraded to the previous version. The database schema is always compatible with the latest version of the platform. Do the backup of the database before any upgrade!
+:::
+
+## Setup database manually
+
+Although not recommended, it is possible to setup the database manually. You will need to take care of the database schema and all the migrations yourself.
+Some of the migrations are more complex and require migration in the code as well. This is another reason why it is recommended to use the platform to handle the database setup.
+
+Each microservice maintains its own database schema and migrations in the GitHub repository. The location of the schema and migrations depends on technology.
+
+The migrations for the Core microservice are located in the [`db/migration`](https://github.com/3KeyCompany/CZERTAINLY-Core/tree/master/src/main/resources/db/migration) directory.
+Depending on your setup, you might be required to initialize the database manually using following additional steps:
 
-- Create the tables using the schema file located in the `db` directory
+Create the tables using the schema file located in the `db` directory:
 ```bash
-psql -h localhost -U czertainlyuser -d czertainlydb < ./core/01_tables.sql
+psql -h localhost -U czertainlyuser -d czertainlydb < db/migration/*.sql
 ```
 
-- Load initial data required by the platform using the file located in the `db` directory
-```bash
-psql -h localhost -U czertainlyuser -d czertainlydb < ./core/02_data.sql
+If you are planning to deploy [Keycloak](../20-integration-guides/10-keycloak/01-overview.md) you need to create schema for it:
```

Comment:
@semik I would remove Keycloak schema from the database setup, because it is an optional component. The database setup should be about how to create and maintain database for the core platform. We also do not have full control about the database setup of the connectors that can be written in any language and any technology, it is similar approach.

Reply:
Imagine yourself in role of DB administrator. You need clear and easy to use information how to prepare database for CZERTAINLY. That is  reason why I'm trying to put info about Keycloak requirements into this section. As far I know, it is only component which behave differently. I tried to rephrase this section to be more clear. Does this help?